### PR TITLE
[Pipeline] filter trigger

### PR DESCRIPTION
**Pull request by J-shang**
### Description ###
The filter will check the files setting in `filter.modified.globs`, if the files are modified, the following `stage: test` will run.

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: pipelines/fast-test.yml [hunk 1]
```diff
@@ -103,10 +115,29 @@ stages:
...
+  jobs:
+  - job: check
+    pool:
+      vmImage: ubuntu-latest
+    steps:
+    - checkout: azpfilter
+      persistCredentials: true
+    - task: NodeTool@0
+      inputs:
+        versionSpec: '16.x'
```
Comment (liuzhe-lz) [inline pipelines/fast-test.yml:128]: We already have this task in templates/config_version.yml

File: pipelines/full-test-hpo.yml [hunk 2]
```diff
@@ -1,38 +1,71 @@
...
+      inputs:
+        versionSpec: '16.x'
+    - script: npm ci
+      displayName: NPM Install
+    - script: node src/main.mjs
+      name: execution
+      displayName: Execution
+
+- stage: test
+  # dependencies.$(StageName).outputs['$(JobName).$(TaskName).$(VariableName)']
```
Comment (ultmaster) [inline pipelines/full-test-hpo.yml:39]: What is this?

File: pipelines/full-test-nas.yml [hunk 3]
```diff
@@ -1,24 +1,57 @@
...
+  repositories:
+    - repository: azpfilter
+      type: github
+      name: microsoft/azure-pipeline-filter
+      endpoint: github-filter-connection
+
+variables:
+  filter.modified.globs: 'examples/nas/**,nni/algorithms/nas/**,nni/nas/**,nni/retiarii/**'
+  filter.prbody.heading: '#### Test Options'
+  filter.prbody.optionIndex: 0
```
Comment (ultmaster) [inline pipelines/full-test-nas.yml:18]: All 0?

File: pipelines/fast-test.yml [hunk 4]
```diff
@@ -103,10 +115,29 @@ stages:
         yarn --cwd ts/webui sanity-check
       displayName: Sanity check (WebUI)
 
+- stage: filter
+  jobs:
+  - job: check
+    pool:
+      vmImage: ubuntu-latest
+    steps:
```
Comment (ultmaster) [inline pipelines/fast-test.yml:123]: Seems these steps have repeated several times. Maybe put them into a template?

File: pipelines/full-test-hpo.yml (hunk 2 above)
Comment (J-shang) [reply pipelines/full-test-hpo.yml:39]: we can get output in one stage by this pattern
here we use:
`dependencies.filter.outputs['check.execution.skipsubsequent']`

File: pipelines/full-test-nas.yml (hunk 3 above)
Comment (J-shang) [reply pipelines/full-test-nas.yml]: updated template

### Description ###

#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### How to test ###

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**